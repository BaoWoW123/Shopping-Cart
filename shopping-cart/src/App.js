import "./styles/app.css";
import React, { useState } from "react";
import Header from "./components/Header";
import Main from "./components/Main";
import Footer from "./components/Footer";
import products from "./components/images";

const App = () => {
  let [items, setItems] = useState(products);
  let [currentProduct, setCurrentProduct] = useState();
  let [cart, setCart] = useState([]);

  const sortItems = (e) => {
    let tempArr;
    e.target.className === "showAll"
      ? (tempArr = products)
      : (tempArr = products.filter(
          (product) => product.type === e.target.className
        ));
    setItems((items = tempArr));
  };

  const selectProduct = (e) => {
    setCurrentProduct((currentProduct = e));
    return currentProduct;
  };

  const addToCart = async (item) => {
    //async for state update before test
    await setCart((prevState) => (cart = [...prevState, item]));
  };

  return (
    <div className="app">
      <Header />
      <Main
        props={(e) => sortItems}
        items={items}
        selectProduct={selectProduct}
        currentProduct={currentProduct}
        addToCart={addToCart}
        cart={cart}
      />
      <Footer />
    </div>
  );
};

export default App;
