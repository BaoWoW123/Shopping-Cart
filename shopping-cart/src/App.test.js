import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import App from "./App";
import { BrowserRouter } from "react-router-dom";

describe("initial render", () => {
  test("On home page", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("link to shop page renders shop", async () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    const shopBtn = container.querySelector(".shopLink");
    await userEvent.click(shopBtn);
    expect(screen.getByRole("heading", { name: "Products" }).textContent).toBe(
      "Products"
    );
  });
});

describe("update render after userEvents", () => {
  test("Click on Cart buttons renders cart", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    const cartBtn = screen.getByAltText("cart image");
    userEvent.click(cartBtn);
    expect(screen.getByRole("heading", { name: "Cart" }).textContent).toBe(
      "Cart"
    );
  });
});
