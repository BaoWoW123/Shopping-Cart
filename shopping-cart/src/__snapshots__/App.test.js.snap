// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render On home page 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="header"
    >
      <h2>
         Boulder Shop
      </h2>
      <a
        class="homeLink"
        href="/"
      >
        <h2>
          Home
        </h2>
      </a>
      <a
        class="shopLink"
        href="/shop"
      >
        <h2>
          Shop
        </h2>
      </a>
      <input
        class="searchBar"
        label="searchBar"
        placeholder="Search for..."
        type="text"
      />
      <button
        class="cartBtn"
      >
        <img
          alt="cart image"
          src="cart.png"
        />
      </button>
    </div>
    <div
      class="main"
    >
      <div
        class="home"
      >
        <div
          class="homeHeader"
        >
          <h1>
            Everything you need to climb
          </h1>
        </div>
        <a
          class="homeShopLink"
          href="/shop"
        >
          <h2>
            Shop
          </h2>
        </a>
      </div>
      <div
        class="cart"
      >
        <h1>
          Cart
        </h1>
        <div>
          Cartitem1
        </div>
        <div>
          Cartitem2
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <a
        class="githubLink"
        href="https://github.com/BaoWoW123/Shopping-Cart"
        target="_blank"
      >
        Check out the code here!
      </a>
    </div>
  </div>
</div>
`;
